frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -598,7 +597,8 @@ function ReaderBookmark:onShowBookmark(match_table)
         self.select_mode = not self.select_mode
         if self.select_mode then
             self.select_count = 0
-            bm_menu:setTitleBarIconAndText("check")
+            bm_menu:setTitleBarLeftIcon("check")
+            UIManager:setDirty(self.show_parent, "ui")
```

Comment:
Shouldn't this be `UIManager:setDirty(bm_menu.show_parent, "ui")` ~~or `UIManager:setDirty(self, "ui")`~~ - as I think there is no self.show_parent. (oh all this bm_menu stuff is confusing :)
But, isn't TitleBar:setLeftIcon() already taking care of the refresh ?
(And you don't do it below in the next chunk.)

Reply:
`setLeftIcon` does it in all cases. Except long-press on left icon to enable select mode.
Long-press to exit select mode changes the icon (I think because we have switchItemTable before it to clear dimmed selections, so more regions are updated).

Reply:
I tested just tapping on the iconm, `setLeftIcon` doesn't update the screen.
The icon is changed only after updating other regions/dialogs.

Reply:
What if you disable Flash button and ui, or use `left_icon_allow_flash=false`?
Just to know if it's again flashing related, or if it is something else like show_parent bad.

Reply:
In CloudStorage `setLeftIcon` changes the icon.
So it is some special Bookmarks issue that requires additional `setDirty`.

Reply:
It doesn't depend on flash_ui. There is no flashing when long-pressing on the buttons.

Reply:
```diff
    bm_menu.close_callback = function()
        UIManager:close(self.bookmark_menu)
    end

    bm_menu.show_parent = self.bookmark_menu
+   bm_menu.title_bar.show_parent = self.bookmark_menu
```
add this, and it works for me - but it's not the cleaner fix...
We wrap bm_menu (Menu) into self.bookmark_menu (CenterContainer), and we give this CenterContainer to UIManager:show(). But we build Menu and TitleBar before we have self.bookmark_menu  that we could give as show_parent.

So, possible 2 fixes needed:

In Menu, when doing :self.title_bar = TitleBar:new{
```diff
- show_parent = self,
+ show_parent = self.show_parent or self
```
as Menu may be given already a show_parent, so use it - otherwise Menu is assumed to be the top window and so we use self.

And rework how ReaderBookmark build it:
```diff
@@ -483,2 +483,6 @@ function ReaderBookmark:onShowBookmark(match_table)

+    self.bookmark_menu = CenterContainer:new{
+        dimen = Screen:getSize(),
+        covers_fullscreen = true, -- hint for UIManager:_repaint()
+    }
     local bm_menu = Menu:new{
@@ -494,4 +498,3 @@ function ReaderBookmark:onShowBookmark(match_table)
         line_color = show_separator and Blitbuffer.COLOR_DARK_GRAY or Blitbuffer.COLOR_WHITE,
-        has_extra_button = true,
-        extra_button_icon = "appbar.menu",
+        title_bar_left_icon = "appbar.menu",
         on_close_ges = {
@@ -506,10 +509,7 @@ function ReaderBookmark:onShowBookmark(match_table)
             }
-        }
+        },
+        show_parent = self.bookmark_menu,
     }
+    table.insert(self.bookmark_menu, bm_menu)

-    self.bookmark_menu = CenterContainer:new{
-        dimen = Screen:getSize(),
-        covers_fullscreen = true, -- hint for UIManager:_repaint()
-        bm_menu,
-    }

@@ -827,3 +831,2 @@ function ReaderBookmark:onShowBookmark(match_table)

-    bm_menu.show_parent = self.bookmark_menu
```

Reply:
I'll try, thanks.

Reply:
It does work.